homeassistant/components/statistics/sensor.py

```diff
@@ -181,9 +188,21 @@ def __init__(
             STAT_BUFFER_USAGE_RATIO: None,
             STAT_SOURCE_VALUE_VALID: None,
         }
-        self._state_characteristic_fn = getattr(
-            self, f"_stat_{self._state_characteristic}"
-        )
+        if self.is_binary and self._state_characteristic not in STATS_BINARY_SUPPORT:
+            raise vol.error.ValueInvalid(
```

Comment:
This is not the correct place to raise, we should raise from the `PLATFORM_SCHEMA` by adding a custom validator which does this check.

This is done in many places, for `PLATFORM_SCHEMA` in example homeassistant/components/mqtt/fan.py

Reply:
Thanks for the hint. I was sure this is not the right way but didn't find a good example. Will look into this

Reply:
Done and test-cased.